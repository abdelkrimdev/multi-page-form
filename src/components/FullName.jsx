import React from 'react'
import PropTypes from 'prop-types'
import { Input, Icon } from 'antd'

const FullName = ({ onChange }) => {
  return (
    <Input
      name='fullName'
      placeholder='Peter Parker'
      prefix={<Icon type='user' style={{ color: 'rgba(0,0,0,.25)' }} />}
      onChange={onChange}
      size='large'
    />
  )
}

FullName.propTypes = {
  onChange: PropTypes.func
}

export default FullName
