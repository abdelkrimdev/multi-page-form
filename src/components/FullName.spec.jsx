import React from 'react'
import { shallow } from 'enzyme'

import FullName from './FullName'

describe('full name component', () => {
  it('should render without crashing.', () => {
    const wrapper = shallow(<FullName />)

    expect(wrapper).toBeDefined()
  })

  it('should render correctly.', () => {
    const wrapper = shallow(<FullName />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should set the onChange property.', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(<FullName onChange={handleChange} />)

    wrapper.prop('onChange')()

    expect(handleChange).toBeCalled()
  })
})
