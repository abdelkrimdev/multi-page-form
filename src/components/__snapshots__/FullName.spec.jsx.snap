// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full name component should render correctly. 1`] = `
<Input
  disabled={false}
  name="fullName"
  placeholder="Peter Parker"
  prefix={
    <Icon
      style={
        Object {
          "color": "rgba(0,0,0,.25)",
        }
      }
      type="user"
    />
  }
  prefixCls="ant-input"
  size="large"
  type="text"
/>
`;
